dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/config/WebClientMetricsFilter.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.justice.laa.crime.dces.integration.config;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+import org.springframework.web.reactive.function.client.ExchangeFunction;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.Objects;
+
+public class WebClientMetricsFilter implements ExchangeFilterFunction {
+
+    private final MeterRegistry meterRegistry;
+    private final String metricsName;
+
+    public WebClientMetricsFilter(MeterRegistry meterRegistry, String metricsName) {
```

Comment:
Just curious, why not use @RequiredArgsConstructor here, and also in MaatApiWebClientFactory?